core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -254,8 +254,8 @@ private static Types.MapType mapFromJson(JsonNode json) {
   }
 
   public static Schema fromJson(JsonNode json) {
-    Type type  = typeFromJson(json);
-    Preconditions.checkArgument(type.isNestedType() && type.asNestedType().isStructType(),
+    Type type = typeFromJson(json);
+    Preconditions.checkArgument(type != null && type.isNestedType() && type.asNestedType().isStructType(),
```

Comment:
Could you open a separate PR for this change?

Reply:
Ah, of course. The Python changes are related, so I figured it would be okay to do it in one PR. I've opened https://github.com/apache/iceberg/pull/5291 with solely the Java change. I'll close this one and include the Python changes in the REST Catalog PR.